src/modelmanager.cpp

```diff
@@ -984,7 +984,7 @@ Status ModelManager::reloadModelWithVersions(ModelConfig& config) {
     }
 
     if (versionsToReload->size() > 0) {
-        reloadModelVersions(model, fs, config, versionsToReload, versionsFailed);
+        blocking_status = reloadModelVersions(model, fs, config, versionsToReload, versionsFailed);
```

Comment:
You are overriding blocking status from adding model version. If reloading succeeds and there are any versions to reload then you may return OK while adding some versions failed.

Reply:
done